### PR TITLE
Fix messaging when Maurice steals a weapon (Salmon)

### PR DESCRIPTION
**Pull request by nlavsky**
When Maurice steals a weapon, the games says
`  Maurice steals <weapon>...`
`  Maurice wields <weapon>...`
`  It bursts into flame!/Its appearance distorts for a moment./etc.`
even though the weapon always goes into his alternative weapon
slot (unless Maurice's disarmed).

The same happens if you give an "alt"-inscribed weapon to a
monster via the xo wizard command.

Fix this, and also simplify monster::equip_weapon().

### CODE REVIEW

**Comment by stenellad** [comment]
To anyone else that reads through this, the bugfix is at the bottom at line 6440. The rest of the changes aren't actually changes at all, just removal of the silly if(msg) wrapping the entire function.

**Comment by stenellad** [comment]
Looking at the new changes:
`equip_weapon`
`equip_armour`
`equip_jewellery`
are only ever called within `equip(). `

`equip()` is called at lines: (and never outside of monster.cc)

1170
1195
1248
2155
6440

All but 6440 seem to have been handled as expected (switching from passing the boolean msg to the function to an if statement before the function call). 

6440 has been changed to include the bugfix from the original commit, making it from always true to 
`if(mslot != MSLOT_ALT_WEAPON || mons_wields_two_weapons(*this))`

tl;dr looks good to me.
